roles/spark-standalone-install/tasks/main.yml

```diff
@@ -0,0 +1,137 @@
+---
+  - name: spark | user | ssh-key | probe
+    become_user: "{{ spark_user }}"
+    shell: cat ~/.ssh/id_rsa.pub
+    register: ssh_key_probe
+
+  - name: spark | user | ssh-key | record
+    set_fact:
+      ssh_key: "{{ ssh_key_probe.stdout }}"
+
+  - name: spark | user | key exchange
+    become_user: "{{ spark_user }}"
+    authorized_key:
+      user: "{{ spark_user }}"
+      key: "{{ hostvars[item]['ssh_key'] }}"
+    with_items: "{{ play_hosts }}"
```

Comment:
This role would be better served including the `ssh-key-exchange` role as a dependent role.  Duplicating one line in user playbooks is not ideal, I admit, but better than duplicating code among roles, imo.

As a side note, I just found out about the `play_hosts` [ansible variable](http://docs.ansible.com/ansible/playbooks_variables.html), which would make for a very nice default for all of those `XYZ_ansible_group` parameters.  Incorporating that variable into all the roles that need a group would solve your duplication issue as well as reduce a lot of clutter in the other roles.